src/peer_manager.rs

```diff
@@ -315,11 +367,179 @@ impl PeerManager {
     }
 
     /// Wraps the routing table function of the same name and maps `XorName`s to `PublicId`s.
-    pub fn expect_add_to_our_group(&self,
-                                   expected_name: &XorName,
-                                   our_public_id: &PublicId)
-                                   -> Result<Vec<Group>, RoutingTableError> {
-        self.routing_table.validate_joining_node(expected_name)?;
+    ///
+    /// Returns: (true, our_group_list)     if needs to carry out resource proof evaluate
+    ///          (false, empty_list)        if doesn't need to carry out resource proof evaluate
+    pub fn expect_join_our_group(&mut self,
+                                 expected_name: &XorName,
+                                 our_public_id: &PublicId)
+                                 -> ExpectJoinResult {
+        if let Some((name, now, _)) = self.node_candidate {
+            if name == *expected_name ||
+               now.elapsed() < Duration::from_secs(RESOURCE_PROOF_APPROVE_TIMEOUT_SECS) {
+                return Err(RoutingTableError::AlreadyExists);
+            }
+        }
+
+        let (prefix, names) = self.routing_table.expect_join_our_group(expected_name)?;
+
+        if names.len() > self.routing_table.min_group_size() {
+            let mut rng = SeededRng::new();
```

Comment:
moving this comment to https://maidsafe.atlassian.net/browse/MAID-1840
so it can be fully discussed in the production code of using this seed.